### DISCUSSION TITLE
2 way video not working on separate networks

### DISCUSSION THREAD

**Discussion by Cartisim** [stasel/WebRTC-iOS#152] [General]
I seem to be experiencing issues in establishing a 2 way video call when the clients (iPhone A & iPhoneB) are not on the same network as each other. I have implemented a production ready signaling server and I have tested both Google's  STUN servers and my own stun servers, yet the connection cannot be made. During the ICE exchange both clients are sending, receiving and adding the needed candidates to the intended destination. The flow works perfectly if they are on the same network, it is only when they are on separate networks. It is driving me a bit mad, so any suggestions I really appreciate.

Thanks

**Comment by stasel** [comment]
Hi @Cartisim 
To support VoIP traffic between different networks, you will also need to deploy your own TURN server.
A TURN server is acting as a relay for the VoIP traffic between two client which cannot directly establish p2p connection.

I hope that helps

**Reply by Cartisim** [reply]
Can it be done without turn? Can I just send it over the wire on my server?
